### ISSUE TITLE
Conditional auditing possible?

### ISSUE THREAD

**Issue by migu0** [collectiveidea/audited#237]
Awesome gem, thanks for creating this.

We have a model that handles both normal data and data that acts as a cache for performant querying. We only want to audit the former.

Is it possible to do something like this, similar to conditions used for callbacks:

``` ruby
class Availability < ActiveRecord::Base
  audited, if: Proc.new { |availability| availability.time_off? || availability.recurring? }
end
```

**Comment by danielmorrison** [comment]
There isn't that syntax, but there are a handful of ways to turn auditing on/off. 

`availability.save_without_auditing` or you can use a block `availability.without_auditing {...}` There are also corresponding class methods. They all end up setting `Availability.auditing_enabled`so you can also do that directly. 

I like the idea of an `if:` syntax but won't have time to add it myself in the short term. I'm happy to accept PRs though!

**Comment by michaelimstepf** [comment]
Ok thanks. I'll use the block for now. Hopefully one day one of us will have time to add that syntax. If you ever add it, please let me know. Thanks

**Comment by btsai** [comment]
why don'y you merge pr https://github.com/collectiveidea/audited/pull/167?

**Comment by squiter** [comment]
:+1: for PR #167

**Comment by jainankita90** [comment]
I tried using this feature. But no luck 
Here is my code


Version 1

```ruby

class Demo
  audited :if => :test_update?
  private
  def test_update?
      if self.a == true
        return true
      end
      return false
   end
end
```


Version 2
```ruby
class Demo
  audited :if=> "audit_comment.present?" 
end
```

**Comment by tbrisker** [comment]
fixed in  c1aa93b
